collect_app/src/main/java/org/odk/collect/android/preferences/PreferencesFragment.java

```diff
@@ -129,6 +133,39 @@ public boolean onPreferenceChange(Preference preference, Object newValue) {
         }
     }
 
+    private void initLanguagePrefs() {
+        final ListPreference pref = (ListPreference) findPreference(KEY_LANGUAGE);
+        final LocaleHelper localeHelper = new LocaleHelper();
+        TreeMap<String, String> languageList = localeHelper.getEntryListValues();
+        pref.setEntryValues(languageList.values().toArray(new String[0]));
+        pref.setEntries(languageList.keySet().toArray(new String[0]));
+
+        if (pref != null) {
```

Comment:
Is there a reason the null checking happens here, but we already start calling pref.<method> before this check? If pref is null, this one will never gets executed because pref.setEntryValues would already throwing NPE.